lib/python/ensembl/brc4/runnable/dump_stable_ids.py

```diff
@@ -0,0 +1,420 @@
+        for old_id in from_list:
+            if not old_id or old_id not in from_list: continue
+            event = StableIdEvent([old_id], from_list[old_id])
+            (event, from_list, to_list) = self.extend_event(event, from_list, to_list)
+            event.add_pairs(pairs)
+            events.append(event)
+        
+        # Remaining events should only be new genes
+        for new_id in to_list:
+            if not new_id: continue
+            event = StableIdEvent(to_list[new_id], [new_id])
+            event.add_pairs(pairs)
+            events.append(event)
+        
+        stats = {}
+        for event in events:
+            name = event.get_name()
+            if not name in stats:
+                stats[name] = 1
+            else:
+                stats[name] += 1
+        
+        for stat in stats:
+            print(f"\t{stat} = {stats[stat]}")
+
+        return events
+    
+    def extend_event(self, event: StableIdEvent, from_list: Dict[str, List[str]], to_list: Dict[str, List[str]]):
+        extended = True
+
+        while(extended):
+            extended = False
+            
+            # Extend the group in the to ids
+            for to_id in event.to_list:
+                if to_id in to_list:
+                    to_from_ids = to_list[to_id]
+                    # Add to the from list?
+                    for to_from_id in to_from_ids:
+                        if not to_from_id in event.from_list:
+                            event.add_from(to_from_id)
+                            extended = True
+            
+            # Extend the group in the from ids
+            for from_id in event.from_list:
+                if from_id in from_list:
+                    from_to_ids = from_list[from_id]
+                    # Add to the to list?
+                    for from_to_id in from_to_ids:
+                        if not from_to_id in event.to_list:
+                            event.add_to(from_to_id)
+                            extended = True
+            
+            # Clean up
+            from_list = {from_id: from_list[from_id] for from_id in from_list if from_id not in event.from_list}
+            to_list = {to_id: to_list[to_id] for to_id in to_list if to_id not in event.to_list}
+        
+        return (event, from_list, to_list)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Dump the stable ids history from a core db')
+
+    parser.add_argument('--host', type=str, required=True, help='Server hostname')
+    parser.add_argument('--port', type=str, required=True, help='Server port')
+    parser.add_argument('--user', type=str, required=True, help='Server user')
+    parser.add_argument('--password', type=str, help='Server password')
+    parser.add_argument('--dbname', type=str, required=True, help='Database name')
+    parser.add_argument('--output_file', type=str, required=True, help='Output file')
+
+    # Optional
+    args = parser.parse_args()
```

Comment:
Same as before about moving this to `if __name__ == "__main__":` for CI/CD purposes.